packages/phone-number-privacy/common/src/utils/authentication.ts

```diff
@@ -37,7 +37,7 @@ export async function authenticateUser(
     try {
       registeredEncryptionKey = await getDataEncryptionKey(signer, contractKit, logger)
     } catch (error) {
-      logger.warn('Assuming request is authenticated')
+      logger.error(ErrorMessage.FAILURE_TO_GET_DEK) // TODO(2.0.0) add monitoring / alerting for this
```

Comment:
per the discussion:
- can we double check that there is no way (other than a provider error) that this fails open? i.e. `getDataEncryptionKey` doesn't throw an error if one isn't set, if the account isn't valid, etc.?
- can you add a comment that this should only fail open on full node error?
- can we explore making fail open logic configurable by env variable (so that in case we notice someone attacking the system during a longer outage, we can ask signers to change their signer env vars instead of requiring a release & redeploy?)

Reply:
configurable fail open logic definitely does not need to be within the scope of this PR but could be in a separate ticket, as you see fit